inc/class-config.php

```diff
@@ -0,0 +1,163 @@
+<?php
+/**
+ * Core platform config loader.
+ *
+ * @package hm-platform-core
+ */
+
+namespace HM\Platform;
+
+class Config {
+
+	/**
+	 * Primary config array.
+	 *
+	 * @var array
+	 */
+	protected static $config = [];
+
+	/**
+	 * Retrieve the configuration for HM Platform.
+	 *
+	 * The configuration is defined by merging the defaults set by modules
+	 * with any overrides present in composer.json.
+	 *
+	 * @return array Configuration data.
+	 */
+	static function get() : array {
+		if ( self::$config ) {
+			return self::$config;
+		}
+
+		self::$config = self::get_merged_defaults_and_customizations();
```

Comment:
Thinking we should pass this through a filter so anyone to make modifications to.

Reply:
I was thinking of filtering late when the config actually gets used but can see future use cases for this 👍